homeassistant/components/fireservicerota/sensor.py

```diff
@@ -0,0 +1,129 @@
+
+    async_add_entities(
+        [IncidentsSensor(coordinator, entry)],
+    )
+
+
+class IncidentsSensor(RestoreEntity, Entity):
+    """Representation of FireServiceRota incidents sensor."""
+
+    def __init__(self, coordinator, entry):
+        """Initialize."""
+        self._coordinator = coordinator
+        self._entry_id = entry.entry_id
+        self._unique_id = entry.unique_id
+
+        self._state = None
+        self._state_attributes = {}
+
+    @property
+    def name(self) -> str:
+        """Return the name of the sensor."""
+        return "Incidents"
+
+    @property
+    def icon(self) -> str:
+        """Return the icon to use in the frontend."""
+        return "mdi:fire-truck"
+
+    @property
+    def state(self) -> str:
+        """Return the state of the sensor."""
+        return self._state
+
+    @property
+    def unique_id(self) -> str:
+        """Return the unique ID for this sensor."""
+        return f"{self._unique_id}_Incidents"
+
+    @property
+    def should_poll(self) -> bool:
+        """No polling needed."""
+        return False
+
+    @property
+    def device_state_attributes(self) -> object:
+        """Return available attributes for sensor."""
+        attr = {}
+        data = self._state_attributes
+
+        if data:
+            for value in (
+                "trigger",
+                "created_at",
+                "message_to_speech_url",
+                "prio",
+                "type",
+                "responder_mode",
+                "can_respond_until",
+            ):
+                if data.get(value):
+                    attr[value] = data[value]
+
+            if "address" in data:
+                for address_value in (
+                    "latitude",
+                    "longitude",
+                    "address_type",
+                    "formatted_address",
+                ):
+                    if address_value in data["address"]:
+                        attr[address_value] = data["address"][address_value]
+
+            attr[ATTR_ATTRIBUTION] = ATTRIBUTION
+            return attr
+
+    async def async_added_to_hass(self) -> None:
+        """Handle entity which will be added."""
+        await super().async_added_to_hass()
+
+        state = await self.async_get_last_state()
+        if state:
+            self._state = state.state
+            self._state_attributes = state.attributes
+            _LOGGER.debug("Restored entity 'Incidents' state to: %s", self._state)
+
+        self.async_on_remove(self._coordinator.stop_listener)
+
+        self.async_on_remove(
+            async_dispatcher_connect(
+                self.hass,
+                f"{FIRESERVICEROTA_DOMAIN}_{self._entry_id}_update",
+                self.async_update,
+            )
+        )
+
+    async def async_update(self) -> None:
+        """Update using FireServiceRota data."""
+        if not self.enabled:
+            return
```

Comment:
This should not be needed IIRC, since if it is disabled it should be removed

Reply:
Removed

Reply:
Just test it properly so I didn't lie to you :)

Reply:
This is a funny one, if I disable the sensor entity via the GUI, it's removed, the web-socket listener is stopped, so self.async_on_remove() code is called, and the integration still shows 1 entity on it's card. but it's nowhere to be found anymore, looks more like a HA bug, or maybe I miss some hooks in my code...

Reply:
in some cases hass reloads the integration after 30 seconds, check so that doesn't happen here.

Reply:
Found it under 'disabled entities' (after I found the filter settings) If I enabled it again it becomes available after a short while and web-socket listener and dispatcher start to work, magic!

Reply:
It works correctly without the check, so all requests are handled I think.